### ISSUE TITLE
[Portal] renders children on a detached DOM node causing measuring and autoFocus to fail

### ISSUE THREAD

**Issue by Swiftwork** [primer/react#1809]
**Describe the bug**
The current way the [Portal](https://github.com/primer/react/blob/main/src/Portal/Portal.tsx) is written, mounts its children on a detached node. This causes children that need to use `autoFocus` or measure themselves to fail. 

**To Reproduce**
1. Example below **will not** be able to auto focus the input.
```js
export const TestDialog = () => {
  return (
    <Dialog2>
      <input autoFocus />
    </Dialog2>
  );
};
```

2. Example below **will** be able to auto focus the input.
```js
export const TestDialog = () => {
  const [mounted, setMounted] = useState(false);
  useEffect(() => {
    setMounted(true);
  }, []);
  return (
    <Dialog2>
      {mounted && <input autoFocus />}
    </Dialog2>
  );
};
```

**Expected behavior**
Possible change to the way children are rendered, taking into account the different drawbacks detailed here https://github.com/reactjs/reactjs.org/issues/272#issuecomment-468024491 and an update to the documentation outlining which method has been chosen. Potentially introducing a prop if possible to switch between mounting styles.

**Desktop (please complete the following information):**
 - OS: All
 - Browser: All
 - Version: v34.2.0

**Smartphone (please complete the following information):**
 - Device: All
 - OS: All
 - Browser: All
 - Version: v34.2.0
 
 Kind Regards
 Erik Hughes

**Comment by lesliecdubs** [comment]
👋🏻  Hi @Swiftwork, sorry for the late reply here and thanks for filing this!

I'm going to ask our accessibility team to take a look at this and advise on whether we want to implement it. @alliethu what do you think?

**Comment by Swiftwork** [comment]
Not stale

**Comment by lesliecdubs** [comment]
Sorry for the delay on this, @Swiftwork. Are you able to share any details about the desired behavior you're trying to accomplish in your application? It'd also be helpful if you could share a reproduction of the problem; you can fork this sandbox template if you'd like: https://codesandbox.io/s/primer-react-qyyepc. Thanks!
